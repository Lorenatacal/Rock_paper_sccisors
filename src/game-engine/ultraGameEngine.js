export const rules = {
  scissorsrock: 1,
  scissorspaper: 1,
  paperrock: 1,
  rockscissors: -1,
  paperscissors: -1,
  rockpaper: -1,
  rockrock: 0,
  scissorsscissors: 0,
  paperpaper: 0
};

export default function gameWinner(option) {
  return rules[option];
}
