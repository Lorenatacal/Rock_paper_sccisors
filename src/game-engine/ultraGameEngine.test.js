import React, { Component } from "react";
import gameWinner from "./ultraGameEngine";
import { rules } from "./ultraGameEngine";
import toJson from "enzyme-to-json";

test("gameWinner() should retun 1 if we pass 'scissorsrock' as argument", () => {
  expect(gameWinner("scissorsrock")).toEqual(1);
});
test("gameWinner() should return -1 if we pass 'rockscissors' as argument", () => {
  expect(gameWinner("rockscissors")).toEqual(-1);
});
test("gameWinner() should return 0 if we pass 'rockrock' as argument", () => {
  expect(gameWinner("rockrock")).toEqual(0);
});
test("rules to have correct shape", () => {
  expect(rules).toMatchSnapshot();
});
