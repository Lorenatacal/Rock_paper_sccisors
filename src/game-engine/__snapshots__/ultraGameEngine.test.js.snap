// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rules to have correct shape 1`] = `
Object {
  "paperpaper": 0,
  "paperrock": 1,
  "paperscissors": -1,
  "rockpaper": -1,
  "rockrock": 0,
  "rockscissors": -1,
  "scissorspaper": 1,
  "scissorsrock": 1,
  "scissorsscissors": 0,
}
`;
